doc/dev/developer_guide/debugging-gdb.rst

```diff
@@ -0,0 +1,47 @@
+GDB - The GNU Project Debugger
+==============================
+
+`The GNU Project Debugger (GDB) <https://www.sourceware.org/gdb>`_ is
+a powerful tool which allows you to analyze the execution flow.
+GDB can help to find bugs, uncover crash errors or track the
+source code during execution.
+
+GET STARTED WITH GDB
```

Comment:
Perhaps include a link or some instructions on how to install GDB? I found some info here https://www.sourceware.org/gdb/download/, but maybe you know of better info on this topic.

Reply:
Maybe I very pedantic.
But other than the info that there exists a script to debug a docker ceph instance, none of the GDB info here is very useful. To be able to debug the ceph code quite some knowledge of GDB is required. People wanting to debug the code already know all this (and more) information.  
So I would only keep the lines 38 - last line

Reply:
@wjwithagen I might be missing something in your comment, but are you saying that the introductory information here is obvious to many ceph developers and therefore not needed? I personally find it helpful since I'm not familiar with GDB myself.

Reply:
> @wjwithagen I might be missing something in your comment, but are you saying that the introductory information here is obvious to many ceph developers and therefore not needed? I personally find it helpful since I'm not familiar with GDB myself.

That is why I disqualified myself upfront as being pedantic.

There is almost no real information in the first part that is not available thru a search engine when you ask for "debug a running process with GDB" And on that page you will get a lot more information. 
So there are 2 chances:

1. Some told you to do this to help figure out a problem you're having. Then that person will also give you instructions what to do. (Which will be different from what is in the PR)
2. You have written some code that does not work as expected. Then perhaps you will need to debug that with GDB. Now IF you have never done that before, you are in for a steep learning curve debugging multi-threaded code. You'd better read up in many of the howto's on the internet and the gdb manual. And again the PR text will not add much useful.

I don't see us writting about all kinds of other knowledge as well. You are expected to know about GCC (or Clang), Cmake and all kinds of other tools. Same goes for debugging code.
What is on the developers info is all Ceph specific, implementation details, tools/script made specifically for Ceph.
So that is why I suggest that only the last few lines are really of interest.

Reply:
I do agree that this documentation may seem quite general, the idea was to provide guidance for developers who aren't familiar with GDB to start using it while working on Ceph.
At first it will probably help mostly newcomers (GSoC, Outreachy and Individuals). However, as the documentation will evolve it may include more Ceph specific ideas and tips contributed by experienced members.

Reply:
@Matan-B 
I understand what you are saying, and yes debugging is complex. 

However including this does not help anybody.
Just one URL pointing to a good HOWTO would be sufficient. Otherwise you will be copying more and more generic GDB info here. Which IMHO serves no purpose.

**Specific** Ceph debug stuff is however most useful.

Reply:
These are good points. I would like for the part explaining how GDB can be attached to a running process to be kept in addition to the section on Teuthology, as it is Ceph-specific. Otherwise, the breakpoint and text-based user interface sections can be removed since they are not Ceph-specific.

Reply:
Updated, the generic info has been excluded.
Thank you for going over this!

Reply:
I'd still appreciate something like a like to a what GDB can do.
Something like:
    [https://www.geeksforgeeks.org/gdb-command-in-linux-with-examples/](url)

Reply:
> I'd still appreciate something like a like to a what GDB can do. Something like: [https://www.geeksforgeeks.org/gdb-command-in-linux-with-examples/](url)

Do you mean that a URL to this guide should be added?

Reply:
It is your PR, so that is up to you.
From my first reaction I suggested replacing the detailed GDB command list with a pointer to a GDB tutorial.
This is one I found by using Google: `howto use GDB`, and it even has a bit more that the just trivial `list, run, continue, break`. 
Feel free to use any other workable reference